### ISSUE TITLE
Filebeat Logstash module seems to not read "format" config option

### ISSUE THREAD

**Issue by sayden** [elastic/beats#10750]
Discovered by @weltenwort and reproduce by myself.

TLDR: It seems that `format: json` config option in Logstash module in Filebeat is not being used so it is always trying to parse plain input

Steps to reproduce: 
* Launch Elasticsearch.
* Enable Logstash module and set it up to use some input log lines (we used the ones from the Logstash module) a config end up looking like this:
```yml
# Module: logstash
# Docs: https://www.elastic.co/guide/en/beats/filebeat/master/filebeat-module-logstash.html

- module: logstash
  # logs
  log:
    enabled: true

    # Set custom paths for the log files. If left empty,
    # Filebeat will choose the paths depending on your OS.
    var.paths: ["/home/mcastro/go/src/github.com/elastic/beats/filebeat/module/logstash/log/test/*.log"]

    # Convert the timestamp to UTC. Requires Elasticsearch >= 6.1.
    #var.convert_timezone: false
    format: json

  # Slow logs
  slowlog:
    enabled: true
    # Set custom paths for the log files. If left empty,
    # Filebeat will choose the paths depending on your OS.
    var.paths: ["/home/mcastro/go/src/github.com/elastic/beats/filebeat/module/logstash/slowlog/test/*.log"]

    # Convert the timestamp to UTC. Requires Elasticsearch >= 6.1.
    #var.convert_timezone: false
    format: json
```
* Launch Filebeat with `filebeat -e`, no errors are shown here
* Pipelines on ES will only show plain pipelines, so `format: json` config in `logstash.yml` has been ignored and the default format, which is `plain` have been loaded:
```json
{
  "xpack_monitoring_2": {
    "description": "This pipeline upgrades documents from the older version of the Monitoring API to the newer version (6) by fixing breaking changes in those older documents before they are indexed from the older version (2).",
    "version": 7000099,
    "processors": [
      {
        "script": {
          "source": "boolean legacyIndex = ctx._index == '.monitoring-data-2';if (legacyIndex || ctx._index.startsWith('.monitoring-es-2')) {if (ctx._type == 'cluster_info') {ctx._type = 'cluster_stats';ctx._id = null;} else if (legacyIndex || ctx._type == 'cluster_stats' || ctx._type == 'node') {String index = ctx._index;Object clusterUuid = ctx.cluster_uuid;Object timestamp = ctx.timestamp;ctx.clear();ctx._id = 'xpack_monitoring_2_drop_bucket';ctx._index = index;ctx._type = 'legacy_data';ctx.timestamp = timestamp;ctx.cluster_uuid = clusterUuid;}if (legacyIndex) {ctx._index = '<.monitoring-es-6-{now}>';}}"
        }
      },
      {
        "rename": {
          "field": "_type",
          "target_field": "type"
        }
      },
      {
        "set": {
          "field": "_type",
          "value": "doc"
        }
      },
      {
        "gsub": {
          "field": "_index",
          "pattern": "(.monitoring-\\w+-)2(-.+)",
          "replacement": "$16$2"
        }
      }
    ]
  },
  "xpack_monitoring_6": {
    "description": "This is a placeholder pipeline for Monitoring API version 6 so that future versions may fix breaking changes.",
    "version": 7000099,
    "processors": []
  },
  "filebeat-8.0.0-logstash-slowlog-pipeline-plain": {
    "description": "Pipeline for parsing logstash slowlogs in the plain format",
    "on_failure": [
      {
        "set": {
          "field": "error.message",
          "value": "{{ _ingest.on_failure_message }}"
        }
      }
    ],
    "processors": [
      {
        "grok": {
          "field": "message",
          "pattern_definitions": {
            "LOGSTASH_CLASS_MODULE": "[\\w\\.]+\\s*",
            "LOGSTASH_LOGLEVEL": "INFO|ERROR|DEBUG|FATAL|WARN|TRACE"
          },
          "patterns": [
            "\\[%{TIMESTAMP_ISO8601:logstash.slowlog.timestamp}\\]\\[%{LOGSTASH_LOGLEVEL:log.level}\\s?\\]\\[%{LOGSTASH_CLASS_MODULE:logstash.slowlog.module}\\] %{GREEDYDATA:message}"
          ]
        }
      },
      {
        "grok": {
          "field": "logstash.slowlog.module",
          "patterns": [
            "slowlog.logstash.%{WORD:logstash.slowlog.plugin_type}.%{WORD:logstash.slowlog.plugin_name}"
          ]
        }
      },
      {
        "grok": {
          "field": "message",
          "patterns": [
            "{:plugin_params=>%{GREEDYDATA:logstash.slowlog.plugin_params}, :took_in_nanos=>%{NUMBER:event.duration}, :took_in_millis=>%{NUMBER:logstash.slowlog.took_in_millis}, :event=>%{GREEDYDATA:logstash.slowlog.event}}"
          ]
        }
      },
      {
        "rename": {
          "field": "@timestamp",
          "target_field": "event.created"
        }
      },
      {
        "date": {
          "formats": [
            "ISO8601"
          ],
          "ignore_failure": true,
          "field": "logstash.slowlog.timestamp",
          "target_field": "@timestamp"
        }
      },
      {
        "remove": {
          "field": [
            "message",
            "logstash.slowlog.timestamp"
          ]
        }
      },
      {
        "convert": {
          "field": "event.duration",
          "type": "long"
        }
      },
      {
        "convert": {
          "field": "logstash.slowlog.took_in_millis",
          "type": "long"
        }
      }
    ]
  },
  "filebeat-8.0.0-logstash-log-pipeline-plain": {
    "description": "Pipeline for parsing logstash logs in the plain format",
    "on_failure": [
      {
        "set": {
          "field": "error.message",
          "value": "{{ _ingest.on_failure_message }}"
        }
      }
    ],
    "processors": [
      {
        "grok": {
          "field": "message",
          "pattern_definitions": {
            "LOGSTASH_CLASS_MODULE": "[\\w\\.]+",
            "LOGSTASH_LOGLEVEL": "INFO|ERROR|DEBUG|FATAL|WARN|TRACE",
            "GREEDYMULTILINE": "(.|\n)*"
          },
          "patterns": [
            "\\[%{TIMESTAMP_ISO8601:logstash.log.timestamp}\\]\\[%{LOGSTASH_LOGLEVEL:log.level}\\s?\\]\\[%{LOGSTASH_CLASS_MODULE:logstash.log.module}\\s*\\] %{GREEDYMULTILINE:message}"
          ]
        }
      },
      {
        "rename": {
          "field": "@timestamp",
          "target_field": "event.created"
        }
      },
      {
        "date": {
          "field": "logstash.log.timestamp",
          "target_field": "@timestamp",
          "formats": [
            "ISO8601"
          ],
          "ignore_failure": true
        }
      },
      {
        "remove": {
          "field": "logstash.log.timestamp"
        }
      }
    ]
  }
}
```
* Because only plain format is being used, there is an error trying to parse the JSON files with Grok:
![image](https://user-images.githubusercontent.com/4249331/52786058-1b9d6300-305a-11e9-9c1c-1b055962a992.png)

One note more, from what I can see, it seems that it has only loaded the `json` files. Two plain lines are missing in this output.
![image](https://user-images.githubusercontent.com/4249331/52786188-877fcb80-305a-11e9-8396-37c8a7ab71e3.png)

**Comment by ycombinator** [comment]
Evidently you must use `var.format: json` (instead of just `format: json`) in your logstash module config file.

**Comment by weltenwort** [comment]
:man_facepalming: thank you @ycombinator

**Comment by weltenwort** [comment]
So it's actually a mistake in the example code in the docs:

![image](https://user-images.githubusercontent.com/973741/52788707-0cbaae80-3062-11e9-9be7-0550aa7f2fe8.png)

@sayden will create a PR for that

**Comment by weltenwort** [comment]
the aforementioned PR is #10754

**Comment by ycombinator** [comment]
Ah, thanks for finding that @weltenwort and fixing it @sayden.
